src/analysis/syntax.ts

```diff
@@ -1,14 +1,71 @@
 import * as SetOps from "../infra/setops";
 
-// TODO: SYNTAX = { ... }, SYNCAT = new Set(Object.values(SYNTAX))
+export const CLASS_DECLARATION = "ClassDeclaration";
+export const METHOD_DECLARATION = "MethodDeclaration";
+export const CONSTRUCTOR = "Constructor";
+export const BLOCK = "block";
+export const CATCH_CLAUSE = "CatchClause";
+
+export const SYNTAX_LIST = "SyntaxList";
 export const VARIABLE_DECLARATION = "VariableDeclaration";
 export const PARAMETER = "Parameter";
+export const ARROW_FUNCTION = "ArrowFunction";
+
+export const RETURN_STATEMENT = "ReturnStatement";
+
+export const CALL_EXPRESSION = "CallExpression";
+export const BINARY_EXPRESSION = "BinaryExpression";
+export const ARRAY_LITERAL_EXPRESSION = "ArrayLiteralExpression";
+export const OBJECT_LITERAL_EXPRESSION = "ObjectLiteralExpression";
+export const TYPE_ASSERTION_EXPRESSION = "TypeAssertionExpression";
+export const PROPERTY_ACCESS_EXPRESSION = "PropertyAccessExpression";
+export const NEW_EXPRESSION = "NewExpression";
+export const PARANTHESIZED_EXPRESSION = "ParenthesizedExpression";
```

Comment:
This is extremely meticulous.

I would argue though that the AST node types are in effect an enum, hence creating a shadow constant for each value is a bit much. If you are worried about typos, a single union type can help:
```typescript
type ASTNodeType = "ClassDeclaration" | "MethodDeclaration" | ...
```

Reply:
Less about typos, more about me having to reference the types when parsing the code
I'd tried to minimize it as much as possible, but in order to build the PTA graph I had to incorporate some language knowledge

Reply:
Actual strings in switch case / no need for them here